### PR TITLE
Terminate gracefully

### PR DESCRIPTION
**Pull request by andistuder**
At current we receive an error raises on each deploy as the background gets terminated. With this change, I try to catch the `TERM` signal and exit without raising an error. 

log example (Heroku instance) before change: 
```
2015-08-14T14:33:11.440953+00:00 app[worker.1]: [2015-08-14 14:33:11 UTC] Running update_events
2015-08-14T14:33:11.553877+00:00 app[worker.1]: [2015-08-14 14:33:11 UTC] Cached 0 events in 0.109616131 seconds
2015-08-14T15:02:41.954643+00:00 heroku[worker.1]: Stopping all processes with SIGTERM
2015-08-14T14:33:12.360277+00:00 app[worker.1]: rake aborted!
2015-08-14T14:33:12.360513+00:00 app[worker.1]: /app/lib/tasks/jobs.rake:48:in `block (3 levels) in <top (required)>'
2015-08-14T14:33:12.360290+00:00 app[worker.1]: SignalException: SIGTERM
2015-08-14T14:33:12.360510+00:00 app[worker.1]: /app/lib/tasks/jobs.rake:48:in `sleep'
2015-08-14T14:33:12.360515+00:00 app[worker.1]: /app/lib/tasks/jobs.rake:38:in `block (2 levels) in <top (required)>'
2015-08-14T14:33:12.360514+00:00 app[worker.1]: /app/lib/tasks/jobs.rake:38:in `loop'
```
log example (Heroku instance) after change: 
```
2015-08-14T14:55:36.410629+00:00 app[worker.1]: [2015-08-14 14:55:36 UTC] Running update_events
2015-08-14T14:55:36.453401+00:00 app[worker.1]: [2015-08-14 14:55:36 UTC] Cached 0 events in 0.037419371 seconds
2015-08-14T14:55:36.878201+00:00 heroku[worker.1]: State changed from up to down
2015-08-14T14:55:39.800195+00:00 heroku[worker.1]: Stopping all processes with SIGTERM
2015-08-14T14:55:40.518166+00:00 app[worker.1]: Terminating...
2015-08-14T14:55:40.518173+00:00 app[worker.1]: ** [NewRelic][08/14/15 14:55:40 +0000 worker.1 (3)] INFO : Starting Agent shutdown
```

### CODE REVIEW

**Comment by andistuder** [comment]
ping @sgerrand @whamilton42 @mattsoutherden could you review this change? 
I hope address the error here https://app.honeybadger.io/projects/44071/faults/14601041 but am unsure if this works on our production stack.

File: lib/tasks/jobs.rake [hunk 1]
```diff
@@ -35,6 +35,10 @@ namespace :jobs do
 
   desc 'Continuously updates event cache'
   task update_events_loop: :environment do
+    Signal.trap('TERM') do
+      puts 'Terminating...'
```
Comment (sgerrand) [inline lib/tasks/jobs.rake]: @andistuder: Perhaps use `warn` instead of `puts.

**Comment by mattsoutherden** [comment]
The question is: Do you need to do anything when quitting to prevent bad state? If not, the that's fine. If you want to finish the current loop iteratiion, you can do something like:
```ruby
Signal.trap('TERM') do
  puts 'Terminating at end of work loop...'
  $exit = true
end
loop do
  do_some_work()
  exit if $exit
  sleep 5
end
```
NB. Don't use globals. ;)

**Comment by mattsoutherden** [comment]
Btw. How *do* we deploy this app?

**Comment by andistuder** [comment]
@mattsoutherden thanks! 
We deploy with [Capistrano](https://github.com/FundingCircle/shipment_tracker-deployment). The process in question is managed with Supervisor.

**Comment by mattsoutherden** [comment]
Ahh. IC. :neckbeard:

**Comment by dideler** [comment]
Added suggestions. Here's the worker logs from Heroku pre and post changes.

Before
```
heroku[worker.1]: Stopping all processes with SIGTERM
app[worker.1]: rake aborted!
app[worker.1]: SignalException: SIGTERM
app[worker.1]: /app/lib/tasks/jobs.rake:48:in `sleep'
app[worker.1]: /app/lib/tasks/jobs.rake:48:in `block (3 levels) in <top (required)>'
app[worker.1]: /app/lib/tasks/jobs.rake:38:in `loop'
app[worker.1]: /app/lib/tasks/jobs.rake:38:in `block (2 levels) in <top (required)>'
app[worker.1]: Tasks: TOP => jobs:update_events_loop
app[worker.1]: (See full trace by running task with --trace)
heroku[worker.1]: Process exited with status 1
```

After
```
heroku[worker.1]: Stopping all processes with SIGTERM
app[worker.1]: Terminating rake task jobs:update_events_loop...
heroku[worker.1]: Process exited with status 0
```

LGTM :+1:

**Pull request merged**